src/goto-programs/replace_calls.cpp

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************\
+
+Module: Replace calls
+
+Author: Daniel Poetzl
+
+\*******************************************************************/
+
+/// \file
+/// Replace calls
+/// Replaces calls to given functions with calls to other given functions. Needs
+/// to be run after removing function pointer calls and before removing returns.
+
+#include "replace_calls.h"
+
+#include <goto-programs/remove_returns.h>
+
+#include <util/base_type.h>
+#include <util/invariant.h>
+#include <util/irep.h>
```

Comment:
I don't think this is necessary, it will have been included several times already.

Reply:
Hm, I think there's some merit to the idea of explicitely including everything that is used and not relying on transitive includes. If for example A includes B and B includes C, and A uses symbols from C, then B could not be refactored to not include C anymore without also having to change A (also see https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md, section "Allow Refactoring").

Reply:
I do subscribe to this principle, but then you'll have to be consistent about it and include `std_code.h`, `namespace.h` and likely a few more.

Reply:
Ok, cool. Yeah true, I actually need to include several more.

Reply:
I don't see where `irept` is used explicitly here.

Reply:
It uses `irep_idt` which is defined in `irep.h`.